src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/ValuesToStringTransformerTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Tests\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\Extension\Core\DataTransformer\ValuesToStringTransformer;
+
+class ValuesToStringTransformerTest extends \PHPUnit_Framework_TestCase
+{
+    private $transformer;
+
+    protected function setUp()
+    {
+        $this->transformer = new ValuesToStringTransformer(',', true);
+    }
+
+    protected function tearDown()
+    {
+        $this->transformer = null;
+    }
+
+    public function testTransform()
+    {
+        $output = 'a,b,c';
+
+        $this->assertSame($output, $this->transformer->transform(array('a', 'b', 'c')));
+    }
+
+    public function testTransformNull()
+    {
+        $this->assertSame('', $this->transformer->transform(null));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Form\Exception\TransformationFailedException
+     */
+    public function testReverseTransformRequiresAnArray()
+    {
+        $this->transformer->transform('a, b, c');
+    }
+
+    public function testReverseTransform()
+    {
+        $input = 'a, b  ,c ';
+
+        $this->assertSame(array('a', 'b', 'c'), $this->transformer->reverseTransform($input));
+    }
+
+    public function testReverseTransformEmpty()
+    {
+        $input = '';
+
+        $this->assertSame(array(), $this->transformer->reverseTransform($input));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Form\Exception\TransformationFailedException
+     */
+    public function testReverseTransformRequiresAString()
+    {
+        $this->transformer->reverseTransform(array('a', 'b', 'c'));
+    }
+}
```

Comment:
However I had one reserve concerning this approach as with `select`, `radio`, and `checkbox`, choices values are hidden from the user, and we only choose a choice label in the view.

But this is not the case `text` as it displays the choice value as editable and so depends on `choice_value` to be clean and `choice_label` to be set.

Maybe with `widget` as `text` we should normalize `choice_label` to show available choices as they need to be input in the text form input in way to pass validation form.